src/state/swap/hooks.tsx

```diff
@@ -85,8 +85,85 @@ export function useDerivedSwapInfo(
     state: TradeState
   }
   allowedSlippage: Percent
-} {
+  swapIsUnsupported: boolean
+  stablecoinValueDifference: Percent | undefined
+  priceImpactTooHigh: boolean
+  showPriceImpactWarning: boolean
+  largerPriceImpact: Percent | undefined
+  priceImpactSeverity: WarningSeverity | undefined
+  swapFiatValues: { amountIn: number | undefined; amountOut: number | undefined }
+  routeIsLoading: boolean
+  routeIsSyncing: boolean
+  swapState: SwapState
+  fiatValueTradeInput: {
+    data: number | undefined
+    isLoading: boolean
+  }
+  fiatValueTradeOutput: {
+    data: number | undefined
+    isLoading: boolean
+  }
+}
+
+export const DerivedSwapInfoContext = createContext<DerivedSwapInfo>({
+  currencies: {},
+  currencyBalances: {},
+  parsedAmount: undefined,
+  inputError: undefined,
+  trade: {
+    trade: undefined,
+    state: TradeState.LOADING,
+  },
+  allowedSlippage: new Percent(0, 100),
+  swapIsUnsupported: false,
+  stablecoinValueDifference: undefined,
+  priceImpactTooHigh: false,
+  showPriceImpactWarning: false,
+  largerPriceImpact: undefined,
+  priceImpactSeverity: undefined,
+  swapFiatValues: { amountIn: undefined, amountOut: undefined },
+  routeIsLoading: false,
+  routeIsSyncing: false,
+  swapState: {
+    independentField: Field.INPUT,
+    typedValue: '',
+    [Field.INPUT]: {
+      currencyId: undefined,
+    },
+    [Field.OUTPUT]: {
+      currencyId: undefined,
+    },
+    recipient: null,
+  },
+  fiatValueTradeInput: {
+    data: undefined,
+    isLoading: false,
+  },
+  fiatValueTradeOutput: {
+    data: undefined,
+    isLoading: false,
+  },
+})
+
+export const DerivedSwapInfoContextProvider = ({
```

Comment:
this is not a hook - should i move this to a different file in this directory that's just for the Derived State Context ?